generators/kubernetes/templates/registry/_jhipster-registry.yml

```diff
@@ -125,13 +121,13 @@ spec:
             LAST_POD_INDEX=$((${CLUSTER_SIZE} - 1))
             REPLICAS=""
             for i in $(seq 0 $LAST_POD_INDEX); do
-              REPLICAS="${REPLICAS}http://admin:${SECURITY_USER_PASSWORD}@${STATEFULSET_NAME}-${i}.jhipster-registry.${STATEFULSET_NAMESPACE}.svc.cluster.local:8761/eureka/"
+              REPLICAS="${REPLICAS}http://admin:${JHIPSTER_REGISTRY_PASSWORD}@${STATEFULSET_NAME}-${i}.jhipster-registry.${STATEFULSET_NAMESPACE}.svc.cluster.local:8761/eureka/"
               if [ $i -lt $LAST_POD_INDEX ]; then
                 REPLICAS="${REPLICAS},"
               fi
             done
-            export EUREKA_CLIENT_SERVICE_URL_DEFAULTZONE=$REPLICAS
-            echo "EUREKA_CLIENT_SERVICE_URL_DEFAULTZONE=${REPLICAS}"
+            export EUREKA_CLIENT_SERVICEURL_DEFAULTZONE=$REPLICAS
+            echo "EUREKA_CLIENT_SERVICEURL_DEFAULTZONE=${REPLICAS}"
 
```

Comment:
Again I had the same issue with EUREKA_CLIENT_SERVICEURL_DEFAULTZONE that the instances didn't discover them properly. I managed to get it working after changing it to EUREKA_CLIENT_SERVICE_URL_DEFAULTZONE. [More info](https://github.com/jhipster/generator-jhipster/pull/5657)

Also, we can remove this initialisation altogether from the 'command' tag. Anyway we have defined the instances to be discovered in 'application-config' configMap. Ideally it should work without this as the client service-url will be picked up from configMap. I tested this in OpenShift by removing the entire initialisation code and it worked without any issue. Pls check that out. Let me also try to test it in K8s.